src/mapping/AxialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,165 @@
+        _vertexDistances.push_back(distance);
+      }
+    } else {
+      PRECICE_ASSERT(_type == COLLECT);
+      PRECICE_CHECK(output()->vertices().size() == 1, "You can only define an axial geometric multiscale mapping of type spread from a mesh with exactly one vertex.");
+    }
+  } else {
+    PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+    PRECICE_ASSERT(false, "Not yet implemented");
+    PRECICE_DEBUG("Compute conservative mapping");
+  }
+  _hasComputedMapping = true;
+}
+
+void AxialGeoMultiscaleMapping::clear()
+{
+  PRECICE_TRACE();
+  _vertexDistances.clear();
+  _hasComputedMapping = false;
+}
+
+void AxialGeoMultiscaleMapping::mapConservative(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+  PRECICE_ASSERT(false, "Not yet implemented");
+  PRECICE_DEBUG("Map conservative");
+}
+
+void AxialGeoMultiscaleMapping::mapConsistent(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_TRACE();
+
+  const int              inValueDimensions = inData.dataDims;
+  const Eigen::VectorXd &inputValues       = inData.values;
+  Eigen::VectorXd &      outputValues      = outData;
+  // TODO: check if this needs to change when access to mesh dimension is possible
+  const int outValueDimensions = outData.size() / output()->vertices().size();
+
+  int effectiveCoordinate = _axis;
+  PRECICE_ASSERT(effectiveCoordinate == 0 ||
+                     effectiveCoordinate == 1 ||
+                     effectiveCoordinate == 2,
+                 "Unknown multiscale axis type.")
+
+  PRECICE_ASSERT((inputValues.size() / inValueDimensions == static_cast<int>(input()->vertices().size())),
+                 inputValues.size(), inValueDimensions, input()->vertices().size());
+  PRECICE_ASSERT((outputValues.size() / outValueDimensions == static_cast<int>(output()->vertices().size())),
+                 outputValues.size(), outValueDimensions, output()->vertices().size());
+
+  PRECICE_DEBUG("Map consistent");
+  if (_type == SPREAD) {
+    /*
+      3D vertices are assigned a value based on distance from the 1D vertex.
+      Currently, a Hagen-Poiseuille profile determines the velocity value.
+    */
+    PRECICE_ASSERT(input()->vertices().size() == 1);
+    mesh::Vertex &v0      = input()->vertices()[0];
+    size_t const  outSize = output()->vertices().size();
+
+    for (size_t i = 0; i < outSize; i++) {
+      PRECICE_ASSERT(static_cast<int>((i * outValueDimensions) + effectiveCoordinate) < outputValues.size(), ((i * outValueDimensions) + effectiveCoordinate), outputValues.size());
+      outputValues((i * outValueDimensions) + effectiveCoordinate) = 2 * inputValues(effectiveCoordinate) * (1 - (_vertexDistances[i] * _vertexDistances[i]));
```

Comment:
Yes that should be correct. Unless the 2D solver is employing axial-symmetry instead of simulating flow between two (infinitely) extended plates, therefore being the cross section of a 3D cylinder 🤔

Reply:
Added a comment.